test/MUIDataTable.test.js

```diff
@@ -359,6 +358,37 @@ describe('<MUIDataTable />', function() {
     assert.strictEqual(actualResult.prop('label'), 'Name: Joe James');
   });
 
+  it('should displayData consider filterList and customFilter are populated', () => {
+    const customFilterColumns = columns.map(c => {
+      if (c.name === "Name") return {
+        ...c,
+        options: {
+          ...c.options,
+          filterOptions: ["B", "J"],
+          customFilter(name, filters) {
+            const firstLetter=name[0];
+            return firstLetter!==filters[0];
+          }
+        }
+      };
+      return c;
+    });    
+
+    const shallowWrapper = shallow(<MUIDataTable columns={customFilterColumns} data={data} />);
+    const table = shallowWrapper.dive();
+    const instance = table.instance();
+    instance.filterUpdate(0, 'J', 'checkbox');
+    table.update();
+
+    const state = table.state();
+    assert.deepEqual(state.filterList, [['J'], [], [], [], []]);
+
+    const expectedResult = JSON.stringify([
+      { data: ['James, Joe', 'Test Corp', renderCities('Yonkers', { rowIndex: 0 }), 'NY', undefined], dataIndex: 0 },
```

Comment:
I made the modifications in order to avoid breaking changes.

I don't understand what you want in test, can you help me understand writing these test code?

Reply:
The main issue is that we don't really gain anything in this test from asserting on `displayData` as that doesn't confirm that the filters worked properly. I think we can just remove that assertion and keep the one above with the `filterList` as that's how the other filter tests work.